---
title: "Causal effects of _Potassium_ and _Sodium_ on cardiometabolic traits"
author: '[Sander W. van der Laan, PhD](https://swvanderlaan.github.io) | @swvanderlaan'
date: '`r Sys.Date()`'
output:
  html_notebook:
    cache: yes
    code_folding: hide
    collapse: yes
    df_print: paged
    fig.align: center
    fig_caption: yes
    fig_height: 10
    fig_retina: 2
    fig_width: 12
    theme: paper
    toc: yes
    toc_float:
      collapsed: no
      smooth_scroll: yes
  html_document:
    df_print: paged
    toc: yes
mainfont: Helvetica
subtitle: Two Sample MR
editor_options:
  chunk_output_type: inline
---
  
# Setting up the environment.
```{r global_options, include=FALSE}
# further define some knitr-options.
knitr::opts_chunk$set(fig.width = 12, fig.height = 8, fig.path = 'Figures/',
                      eval = TRUE, warning = FALSE, message = FALSE)
```

Clearing the environment in case anything exists.
```{r ClearEnvironment, echo = FALSE}
rm(list = ls())
```

Some general settings.
```{r GeneralSettings}
cat("\nSetting working directory and listing its contents.\n")
# MacBook Pro
ROOT_loc="/Users/swvanderlaan/OneDrive - UMC Utrecht"
PLINK_loc=paste0(ROOT_loc,"/PLINK")
GENOMICS_loc=paste0(ROOT_loc,"/Genomics")

### Generic Locations
AEDB_loc = paste0(GENOMICS_loc, "/AE-AAA_GS_DBs")
LAB_loc = paste0(GENOMICS_loc, "/LabBusiness")
RAWDATA = paste0(ROOT_loc, "/PLINK/analyses/consortia/GWAS_PotassiumSodium/")
RESULTS = paste0(RAWDATA, "/ANALYSIS2018")

PROJECT_loc = paste0(RAWDATA, "/ANALYSIS2018/TwoSampleMR")

### SOME VARIABLES WE NEED DOWN THE LINE
cat("\nDefining phenotypes and datasets.\n")
PROJECTNAME="RESULTS"
TARGET_GENES="PotSod"

cat("\nCreate a new analysis directory, including subdirectories.\n")
# Analysis
ifelse(!dir.exists(file.path(PROJECT_loc, "/",PROJECTNAME)), 
       dir.create(file.path(PROJECT_loc, "/",PROJECTNAME)), 
       FALSE)
ANALYSIS_loc = paste0(PROJECT_loc,"/",PROJECTNAME)

# Plots
ifelse(!dir.exists(file.path(ANALYSIS_loc, "/PLOTS")), 
       dir.create(file.path(ANALYSIS_loc, "/PLOTS")), 
       FALSE)
PLOT_loc = paste0(ANALYSIS_loc,"/PLOTS")

# Output files
ifelse(!dir.exists(file.path(ANALYSIS_loc, "/OUTPUT")), 
       dir.create(file.path(ANALYSIS_loc, "/OUTPUT")), 
       FALSE)
OUT_loc = paste0(ANALYSIS_loc, "/OUTPUT")

cat("\nSetting working directory and listing its contents.\n")
setwd(paste0(PROJECT_loc))
getwd()
list.files()

```


Package-installation function.
```{r Function: installations, echo=FALSE}
install.packages.auto <- function(x) { 
  x <- as.character(substitute(x)) 
  if(isTRUE(x %in% .packages(all.available = TRUE))) { 
    eval(parse(text = sprintf("require(\"%s\")", x)))
  } else { 
    # Update installed packages - this may mean a full upgrade of R, which in turn
    # may not be warrented. 
    # update.install.packages.auto(ask = FALSE) 
    eval(parse(text = sprintf("install.packages(\"%s\", dependencies = TRUE, repos = \"https://cloud.r-project.org/\")", x)))
  }
  if(isTRUE(x %in% .packages(all.available = TRUE))) { 
    eval(parse(text = sprintf("require(\"%s\")", x)))
  } else {
    if (!requireNamespace("BiocManager"))
      install.packages("BiocManager")
    # BiocManager::install() # this would entail updating installed packages, which in turned may not be warrented
    eval(parse(text = sprintf("BiocManager::install(\"%s\")", x)))
    eval(parse(text = sprintf("require(\"%s\")", x)))
  }
}
```

Load those packages.
```{r Setting: loading_packages, echo=FALSE, message=FALSE, warning=FALSE}
# Packages needed for tabulating results, parsing data, or making graphics
install.packages.auto("devtools")
library(devtools) 

install.packages.auto("readr")
install.packages.auto("optparse")
install.packages.auto("tools")
install.packages.auto("dplyr")
install.packages.auto("tidyr")
install.packages.auto("tidylog")
library("tidylog", warn.conflicts = FALSE)
install.packages.auto("naniar")

# To get 'data.table' with 'fwrite' to be able to directly write gzipped-files
# Ref: https://stackoverflow.com/questions/42788401/is-possible-to-use-fwrite-from-data-table-with-gzfile
# install.packages("data.table", repos = "https://Rdatatable.gitlab.io/data.table")
install.packages.auto("data.table")

install.packages.auto("tidyverse")
install.packages.auto("knitr")
install.packages.auto("DT")
install.packages.auto("eeptools")

install.packages.auto("haven")
install.packages.auto("tableone")

install.packages.auto("ggpubr")

# TwoSample MR
# devtools::install_github("MRCIEU/TwoSampleMR")
library("TwoSampleMR")

```

We will create a datestamp and define the Utrecht Science Park Colour Scheme.
```{r Setting: Colors, echo=FALSE}

Today = format(as.Date(as.POSIXlt(Sys.time())), "%Y%m%d")
Today.Report = format(as.Date(as.POSIXlt(Sys.time())), "%A, %B %d, %Y")

### UtrechtScienceParkColoursScheme
###
### WebsitetoconvertHEXtoRGB:http://hex.colorrrs.com.
### Forsomefunctionsyoushoulddividethesenumbersby255.
### 
###	No.	Color			      HEX	(RGB)						              CHR		  MAF/INFO
###---------------------------------------------------------------------------------------
###	1	  yellow			    #FBB820 (251,184,32)				      =>	1		or 1.0>INFO
###	2	  gold			      #F59D10 (245,157,16)				      =>	2		
###	3	  salmon			    #E55738 (229,87,56)				      =>	3		or 0.05<MAF<0.2 or 0.4<INFO<0.6
###	4	  darkpink		    #DB003F ((219,0,63)				      =>	4		
###	5	  lightpink		    #E35493 (227,84,147)				      =>	5		or 0.8<INFO<1.0
###	6	  pink			      #D5267B (213,38,123)				      =>	6		
###	7	  hardpink		    #CC0071 (204,0,113)				      =>	7		
###	8	  lightpurple	    #A8448A (168,68,138)				      =>	8		
###	9	  purple			    #9A3480 (154,52,128)				      =>	9		
###	10	lavendel		    #8D5B9A (141,91,154)				      =>	10		
###	11	bluepurple		  #705296 (112,82,150)				      =>	11		
###	12	purpleblue		  #686AA9 (104,106,169)			      =>	12		
###	13	lightpurpleblue	#6173AD (97,115,173/101,120,180)	=>	13		
###	14	seablue			    #4C81BF (76,129,191)				      =>	14		
###	15	skyblue			    #2F8BC9 (47,139,201)				      =>	15		
###	16	azurblue		    #1290D9 (18,144,217)				      =>	16		or 0.01<MAF<0.05 or 0.2<INFO<0.4
###	17	lightazurblue	  #1396D8 (19,150,216)				      =>	17		
###	18	greenblue		    #15A6C1 (21,166,193)				      =>	18		
###	19	seaweedgreen	  #5EB17F (94,177,127)				      =>	19		
###	20	yellowgreen		  #86B833 (134,184,51)				      =>	20		
###	21	lightmossgreen	#C5D220 (197,210,32)				      =>	21		
###	22	mossgreen		    #9FC228 (159,194,40)				      =>	22		or MAF>0.20 or 0.6<INFO<0.8
###	23	lightgreen	  	#78B113 (120,177,19)				      =>	23/X
###	24	green			      #49A01D (73,160,29)				      =>	24/Y
###	25	grey			      #595A5C (89,90,92)				        =>	25/XY	or MAF<0.01 or 0.0<INFO<0.2
###	26	lightgrey		    #A2A3A4	(162,163,164)			      =>	26/MT
### 
###	ADDITIONAL COLORS
###	27	midgrey			#D7D8D7
###	28	verylightgrey	#ECECEC"
###	29	white			#FFFFFF
###	30	black			#000000
###----------------------------------------------------------------------------------------------

uithof_color = c("#FBB820","#F59D10","#E55738","#DB003F","#E35493","#D5267B",
                 "#CC0071","#A8448A","#9A3480","#8D5B9A","#705296","#686AA9",
                 "#6173AD","#4C81BF","#2F8BC9","#1290D9","#1396D8","#15A6C1",
                 "#5EB17F","#86B833","#C5D220","#9FC228","#78B113","#49A01D",
                 "#595A5C","#A2A3A4", "#D7D8D7", "#ECECEC", "#FFFFFF", "#000000")

uithof_color_legend = c("#FBB820", "#F59D10", "#E55738", "#DB003F", "#E35493",
                        "#D5267B", "#CC0071", "#A8448A", "#9A3480", "#8D5B9A",
                        "#705296", "#686AA9", "#6173AD", "#4C81BF", "#2F8BC9",
                        "#1290D9", "#1396D8", "#15A6C1", "#5EB17F", "#86B833",
                        "#C5D220", "#9FC228", "#78B113", "#49A01D", "#595A5C",
                        "#A2A3A4", "#D7D8D7", "#ECECEC", "#FFFFFF", "#000000")

#ggplot2 default color palette
gg_color_hue <- function(n) {
  hues = seq(15, 375, length = n + 1)
  hcl(h = hues, l = 65, c = 100)[1:n]
}

### ----------------------------------------------------------------------------
```



# Two Sample MR

We performed the generalized summary-data based Mendelian randomization (GSMR) to assess the extent to which cardiometabolic traits are causal to FABP4 levels. 

Here we want to test whether FABP4 levels are causal to cardiometabolic traits using Two Sample Mendelian Randomization (MR) as described [elsewhere](https://mrcieu.github.io/TwoSampleMR/index.html). To this end we will:

- load the data from the potassium/sodium GWAS
- load the results from the cardiometabolic trait using the proxies (r<sup>2</sup> = 1.0) of the top variants in the PotSod GWAS


# Load results

## GWAS

```{r LoadResults: MODEL1 discovery}
M1qc_disc <- fread(paste0(RESULTS, "/ZENODO/meta.GWAS.FABP4.1Gp1.EUR.MODEL1.summaryQC.ELISAonly.QCed.txt.gz"),
                verbose = FALSE, showProgress = TRUE, nThread = 8)

```

### Extract proxies

```{r Proxies}
hits_proxies <- fread(paste0(RESULTS, "/metagwas_fabp4/CROSSTRAIT/variant.crosstraits.txt"),
                verbose = FALSE, showProgress = TRUE, nThread = 8)

```

```{r}
M2qc_disc_subset <- merge(hits_proxies, M2qc_disc, by.x = "VariantID", by.y = "VARIANTID", 
                          sort = FALSE, all.x = TRUE)
M3qc_disc_subset <- merge(hits_proxies, M3qc_disc, by.x = "VariantID", by.y = "VARIANTID", 
                          sort = FALSE, all.x = TRUE)
rm(M2qc_disc, M3qc_disc)
```


The discovery stage proxies.
```{r}
DT::datatable(M1qc_disc_subset)
```
The discovery stage proxies - model 2: model 1 + BMI.
```{r}
DT::datatable(M2qc_disc_subset)
```



# Two Sample MR

One of the assumptions in Mendelian Randomisation (MR) analyses is that the instrument(s) used, should only act through the gene _c.q._ protein expression (_i.e._ the exposure) on the outcome. Therefore, we chose to only include _cis_-acting protein quantitative trait loci (pQTLs). The disadvantage is the loss in power, but the MR does gain specificity as _cis_-pQTLs are less likely to suffer from heterogeneity and pleiotropy.


Let's first prepare our data for Two Sample MR.

## Preparation

### Exposure: FABP4

A data frame of the instruments for an exposure is required. Each line has the information for one variant for one exposure. The minimum information required for MR analysis is the following:

- `SNP` - rs ID
- `beta` - The effect size. If the trait is binary then log(OR) should be used
- `se` - The standard error of the effect size
- `effect_allele` - The allele of the SNP which has the effect marked in beta


Other information that is useful for MR can also be provided:

- `other_allele` - The non-effect allele
- `eaf` - The effect allele frequency
- `Phenotype` - The name of the phenotype for which the SNP has an effect


You can also provide the following extra information:

- `chr` - Physical position of variant (chromosome)
- `position` - Physical position of variant (position)
- `samplesize` - Sample size for estimating the effect size
- `ncase` - Number of cases
- `ncontrol` - Number of controls
- `pval` - The P-value for the SNPâ€™s association with the exposure
- `units` - The units in which the effects are presented
- `gene` - The gene or other annotation for the the SNP


```{r format FABP4}

M1qc_disc_exp_dat <- format_data(M1qc_disc_subset, type="exposure",
                                 header = TRUE,
                                 # phenotype_col = "",
                                 snp_col = "VariantID",
                                 beta_col = "BETA_FIXED",
                                 se_col = "SE_FIXED",
                                 eaf_col = "CAF",
                                 effect_allele_col = "CODEDALLELE",
                                 other_allele_col = "OTHERALLELE",
                                 pval_col = "P_FIXED",
                                 # gene_col = "NEAREST_GENE",
                                 chr_col = "Chr",
                                 pos_col = "BP",
                                 z_col = "Z_FIXED",
                                 samplesize_col = "N_EFF")

M1qc_disc_exp_dat$exposure <- "FABP4 (discovery)"
DT::datatable(M1qc_disc_exp_dat)

M2qc_disc_exp_dat <- format_data(M2qc_disc_subset, type="exposure",
                                 header = TRUE,
                                 # phenotype_col = "",
                                 snp_col = "VariantID",
                                 beta_col = "BETA_FIXED",
                                 se_col = "SE_FIXED",
                                 eaf_col = "CAF",
                                 effect_allele_col = "CODEDALLELE",
                                 other_allele_col = "OTHERALLELE",
                                 pval_col = "P_FIXED",
                                 # gene_col = "NEAREST_GENE",
                                 chr_col = "Chr",
                                 pos_col = "BP",
                                 z_col = "Z_FIXED",
                                 samplesize_col = "N_EFF")

M2qc_disc_exp_dat$exposure <- "FABP4 (discovery, model 2)"
DT::datatable(M2qc_disc_exp_dat)

M3qc_disc_exp_dat <- format_data(M3qc_disc_subset, type="exposure",
                                 header = TRUE,
                                 # phenotype_col = "",
                                 snp_col = "VariantID",
                                 beta_col = "BETA_FIXED",
                                 se_col = "SE_FIXED",
                                 eaf_col = "CAF",
                                 effect_allele_col = "CODEDALLELE",
                                 other_allele_col = "OTHERALLELE",
                                 pval_col = "P_FIXED",
                                 # gene_col = "NEAREST_GENE",
                                 chr_col = "Chr",
                                 pos_col = "BP",
                                 z_col = "Z_FIXED",
                                 samplesize_col = "N_EFF")

M3qc_disc_exp_dat$exposure <- "FABP4 (discovery, model 3)"
DT::datatable(M3qc_disc_exp_dat)

M1qc_repl_exp_dat <- format_data(M1qc_repl_subset, type="exposure",
                                 header = TRUE,
                                 # phenotype_col = "",
                                 snp_col = "VariantID",
                                 beta_col = "BETA_FIXED",
                                 se_col = "SE_FIXED",
                                 eaf_col = "CAF",
                                 effect_allele_col = "CODEDALLELE",
                                 other_allele_col = "OTHERALLELE",
                                 pval_col = "P_FIXED",
                                 chr_col = "Chr",
                                 pos_col = "BP",
                                 z_col = "Z_FIXED",
                                 samplesize_col = "N")

M1qc_repl_exp_dat$exposure <- "FABP4 (replication)"
DT::datatable(M1qc_repl_exp_dat)

```

### Outcome: cardiometabolic traits

We should also parse the outcome data.

The `extract_outcome_data` function returns a table of SNP effects for the requested SNPs on the requested outcomes. The format of the data is similar to the exposure data format, except the main columns are as follows:

- SNP
- beta.outcome
- se.outcome
- samplesize.outcome
- ncase.outcome
- ncontrol.outcome
- pval.outcome
- eaf.outcome
- effect_allele.outcom
- other_allele.outcome
- units.outcome
- outcome
- consortium.outcome
- year.outcome
- pmid.outcome
- id.outcome
- originalname.outcome
- proxy.outcome
- target_snp.outcome
- proxy_snp.outcome
- target_a1.outcome
- target_a2.outcome
- proxy_a1.outcome
- proxy_a2.outcome
- mr_keep.outcome
- data_source.outcome


Earlier we performed a lookup of the 63 proxies across 23 cardiometabolic traits. We load these here.

```{r Crosstrait}
crosstrait <- fread(paste0(RESULTS, "/metagwas_fabp4/CROSSTRAIT/final_unfiltered.tsv"),
                verbose = FALSE, showProgress = TRUE, nThread = 8)
DT::datatable(crosstrait)
```

We load the crosstrait results also directly through the `TwoSampleMR` function.

```{r}
# Note we can simply use the discovery dataset list of SNPs - these are the same as the replication
outcome_dat <- read_outcome_data(snps = M1qc_disc_exp_dat$SNP,
                                 filename = paste0(RESULTS, "/metagwas_fabp4/CROSSTRAIT/final_unfiltered.tsv"),
                                 sep = "\t",
                                 snp_col = "VariantID",
                                 beta_col = "BETA",
                                 se_col = "SE",
                                 effect_allele_col = "EffectAllele",
                                 other_allele_col = "OtherAllele",
                                 eaf_col = "EAF",
                                 pval_col = "P",
                                 samplesize_col = "N",
                                 phenotype_col = "Trait")
```


## Harmonisation

We need to harmonise the data and make sure there are no issues with respect to alleles and strand mismatches. We take a conserbative approach and try to infer positive strand alleles using the allele frequencies for palindromes (_i.e._ A/T-C/G-variants).

```{r Harmonise discovery}
M1qc_disc_dat <- harmonise_data(exposure_dat = M1qc_disc_exp_dat, outcome_dat = outcome_dat)
DT::datatable(M1qc_disc_dat)
```


```{r Harmonise replication}
M1qc_repl_dat <- harmonise_data(exposure_dat = M1qc_repl_exp_dat, outcome_dat = outcome_dat)
DT::datatable(M1qc_repl_dat)
```

```{r Harmonise discovery m2}
M2qc_disc_dat <- harmonise_data(exposure_dat = M2qc_disc_exp_dat, outcome_dat = outcome_dat)
DT::datatable(M2qc_disc_dat)
```

```{r Harmonise discovery m3}
M3qc_disc_dat <- harmonise_data(exposure_dat = M3qc_disc_exp_dat, outcome_dat = outcome_dat)
DT::datatable(M3qc_disc_dat)
```

## Mendelian randomization (MR)

We will perform inverse-variance weighted MR and apply MR Egger to determine the extent of pleiotropy. 

```{r MR discovery}
M1qc_disc_res <- mr(M1qc_disc_dat, method_list=c("mr_egger_regression", "mr_ivw"))

DT::datatable(M1qc_disc_res)
```

```{r MR replication}
M1qc_repl_res <- mr(M1qc_repl_dat, method_list=c("mr_egger_regression", "mr_ivw"))

DT::datatable(M1qc_repl_res)
```

## Sensitivity analyses

### Heterogeneity

We should test for heterogeneity in our analyses. 

```{r Heterogeneity discovery}
M1qc_disc_het <- mr_heterogeneity(M1qc_disc_dat, method_list=c("mr_egger_regression", "mr_ivw"))

DT::datatable(M1qc_disc_het)
```

```{r Heterogeneity replication}
M1qc_repl_het <- mr_heterogeneity(M1qc_repl_dat, method_list=c("mr_egger_regression", "mr_ivw"))

DT::datatable(M1qc_repl_het)
```

### Horizontal pleiotropy

The intercept term in MR Egger regression can be a useful indication of whether directional horizontal pleiotropy is driving the results of an MR analysis. 

```{r Pleiotropy discovery}
M1qc_disc_pleio <- mr_pleiotropy_test(M1qc_disc_dat)

DT::datatable(M1qc_disc_pleio)
```

```{r Pleiotropy replication}
M1qc_repl_pleio <- mr_pleiotropy_test(M1qc_repl_dat)

DT::datatable(M1qc_repl_pleio)
```


## Single SNP analysis

To obtain the MR estimates using each of the individual SNPs we can do the following:

```{r Single discovery}
M1qc_disc_single <- mr_singlesnp(M1qc_disc_dat, 
                                 single_method = "mr_meta_fixed",
                                 all_method = c("mr_ivw", "mr_egger_regression"))

DT::datatable(M1qc_disc_single)
```

```{r Single replication}
M1qc_repl_single <- mr_singlesnp(M1qc_repl_dat,
                                 single_method = "mr_meta_fixed",
                                 all_method = c("mr_ivw", "mr_egger_regression"))

DT::datatable(M1qc_repl_single)
```
## Leave-on-out analyses

As these variants are highly correlated, we want to assess if all variant equally are driving the associations. To this end we performed a leave-one-out analysis, where the MR is performed again but leaving out each SNP in turn.

```{r LOO discovery}
M1qc_disc_loo <- mr_leaveoneout(M1qc_disc_dat, 
                                method = mr_ivw)

DT::datatable(M1qc_disc_loo)
```

```{r LOO replication}
M1qc_repl_loo <- mr_leaveoneout(M1qc_repl_dat,
                                method = mr_ivw)

DT::datatable(M1qc_repl_loo)
```

## Visualisation

### Scatters

We can create scatter plots for each trait separately.
```{r Scatter discovery}
library(ggplot2)

p1_disc <- mr_scatter_plot(M1qc_disc_res, M1qc_disc_dat)

for (TRAIT in 1:23){
  trait_print = unique(p1_disc[[TRAIT]]$data$outcome)
  cat(paste0("Processing dataset ", trait_print, " (#",TRAIT,").\n"))
  
  cat("- plotting\n")
  print(p1_disc[[TRAIT]] + theme_minimal())
  
  cat("- saving plots\n\n")
  ggsave(p1_disc[[TRAIT]], 
         file = paste0(PLOT_loc, "/",Today,"MR.results.disc.",trait_print,".pdf"), 
                                    width = 10, height = 10)
  ggsave(p1_disc[[TRAIT]], 
         file = paste0(PLOT_loc, "/",Today,"MR.results.disc.",trait_print,".png"), 
                                    width = 10, height = 10)
  
}

```

```{r Scatter replication}
library(ggplot2)

p1_repl <- mr_scatter_plot(M1qc_repl_res, M1qc_repl_dat)

for (TRAIT in 1:23){
  trait_print = unique(p1_repl[[TRAIT]]$data$outcome)
  cat(paste0("Processing dataset ", trait_print, " (#",TRAIT,").\n"))
  
  cat("- plotting\n")
  print(p1_repl[[TRAIT]] + theme_minimal())
  
  cat("- saving plots\n\n")
  ggsave(p1_repl[[TRAIT]], 
         file = paste0(PLOT_loc, "/",Today,"MR.results.repl.",trait_print,".pdf"), 
                                    width = 10, height = 10)
  ggsave(p1_repl[[TRAIT]], 
         file = paste0(PLOT_loc, "/",Today,"MR.results.repl.",trait_print,".png"), 
                                    width = 10, height = 10)
  
}

```


### Leave-One-Out

We can create Leave-One-Out plots for each trait separately.
```{r LOOPlot discovery}
library(ggplot2)

p2_disc <- mr_leaveoneout_plot(M1qc_disc_loo)

for (TRAIT in 1:23){
  trait_print = unique(p2_disc[[TRAIT]]$data$outcome)
  cat(paste0("Processing dataset ", trait_print, " (#",TRAIT,").\n"))
  
  cat("- plotting\n")
  print(p2_disc[[TRAIT]] + theme_minimal())
  
  cat("- saving plots\n\n")
  ggsave(p2_disc[[TRAIT]], 
         file = paste0(PLOT_loc, "/",Today,"MR.results.disc.loo.",trait_print,".pdf"), 
                                    width = 10, height = 10)
  ggsave(p2_disc[[TRAIT]], 
         file = paste0(PLOT_loc, "/",Today,"MR.results.disc.loo.",trait_print,".png"), 
                                    width = 10, height = 10)
  
}

```


```{r LOOPlot replication}
library(ggplot2)

p2_repl <- mr_leaveoneout_plot(M1qc_repl_loo)

for (TRAIT in 1:23){
  trait_print = unique(p2_repl[[TRAIT]]$data$outcome)
  cat(paste0("Processing dataset ", trait_print, " (#",TRAIT,").\n"))
  
  cat("- plotting\n")
  print(p2_repl[[TRAIT]] + theme_minimal())
  
  cat("- saving plots\n\n")
  ggsave(p2_repl[[TRAIT]], 
         file = paste0(PLOT_loc, "/",Today,"MR.results.repl.loo.",trait_print,".pdf"), 
                                    width = 10, height = 10)
  ggsave(p2_repl[[TRAIT]], 
         file = paste0(PLOT_loc, "/",Today,"MR.results.repl.loo.",trait_print,".png"), 
                                    width = 10, height = 10)
  
}

```


### Funnel plot

We can create funnel plots for each trait separately to assess the asymmetry. This is useful for gauging the reliability of the MR analysis.

```{r Funnel discovery}
library(ggplot2)

p3_disc <- mr_funnel_plot(M1qc_disc_single)

for (TRAIT in 1:23){
  trait_print = unique(p3_disc[[TRAIT]]$data$outcome)
  cat(paste0("Processing dataset ", trait_print, " (#",TRAIT,").\n"))
  
  cat("- plotting\n")
  print(p3_disc[[TRAIT]] + theme_minimal())
  
  cat("- saving plots\n\n")
  ggsave(p3_disc[[TRAIT]], 
         file = paste0(PLOT_loc, "/",Today,"MR.results.disc.funnel.",trait_print,".pdf"), 
                                    width = 10, height = 10)
  ggsave(p3_disc[[TRAIT]], 
         file = paste0(PLOT_loc, "/",Today,"MR.results.disc.funnel.",trait_print,".png"), 
                                    width = 10, height = 10)
  
}

```


```{r Funnel replication}
library(ggplot2)

p3_repl <- mr_funnel_plot(M1qc_repl_single)

for (TRAIT in 1:23){
  trait_print = unique(p3_repl[[TRAIT]]$data$outcome)
  cat(paste0("Processing dataset ", trait_print, " (#",TRAIT,").\n"))
  
  cat("- plotting\n")
  print(p3_repl[[TRAIT]] + theme_minimal())
  
  cat("- saving plots\n\n")
  ggsave(p3_repl[[TRAIT]], 
         file = paste0(PLOT_loc, "/",Today,"MR.results.repl.funnel.",trait_print,".pdf"), 
                                    width = 10, height = 10)
  ggsave(p3_repl[[TRAIT]], 
         file = paste0(PLOT_loc, "/",Today,"MR.results.repl.funnel.",trait_print,".png"), 
                                    width = 10, height = 10)
  
}

```

# FABP4 vs 23 cardiometabolic traits

## Top hits

We will use perfect proxies (r<sup>2</sup> = 0.99 , D' = 1.0) for the two top hits, rs13064760 -6,555 bp from rs2012444 near _PPARG_ and rs16909187 -190 bp from rs16909196 near _FABP4_, given that these show the strongest independent statistical association with circulating serum FABP4 levels. 

We also select rs77878271 near _FABP4_ as this is a functional variant, however, only few proxies exist _and_ were present in crosstrait datasets.

```{r}
M1qc_disc_subset_top <- subset(M1qc_disc_subset, VariantID == "rs13064760" | VariantID == "rs16909187" | VariantID == "rs77878271")

M2qc_disc_subset_top <- subset(M2qc_disc_subset, VariantID == "rs13064760" | VariantID == "rs16909187" | VariantID == "rs77878271")

M3qc_disc_subset_top <- subset(M3qc_disc_subset, VariantID == "rs13064760" | VariantID == "rs16909187" | VariantID == "rs77878271")

M1qc_repl_subset_top <- subset(M1qc_repl_subset, VariantID == "rs13064760" | VariantID == "rs16909187" | VariantID == "rs77878271")

head(M1qc_disc_subset_top)
head(M2qc_disc_subset_top)
head(M3qc_disc_subset_top)
head(M1qc_repl_subset_top)
```

## One-to-Many

We should visualize the causal effect of FABP4 on the 23 cardiometabolic traits.


### Prepare

First we will prepare the data. 
```{r Prepare One2Many}

M1qc_disc_exp_dat_top <- format_data(M1qc_disc_subset_top, type="exposure",
                                 header = TRUE,
                                 # phenotype_col = "",
                                 snp_col = "VariantID",
                                 beta_col = "BETA_FIXED",
                                 se_col = "SE_FIXED",
                                 eaf_col = "CAF",
                                 effect_allele_col = "CODEDALLELE",
                                 other_allele_col = "OTHERALLELE",
                                 pval_col = "P_FIXED",
                                 # gene_col = "NEAREST_GENE",
                                 chr_col = "Chr",
                                 pos_col = "BP",
                                 z_col = "Z_FIXED",
                                 samplesize_col = "N_EFF")

M1qc_disc_exp_dat_top$exposure <- "FABP4 (discovery)"
DT::datatable(M1qc_disc_exp_dat_top)

M2qc_disc_exp_dat_top <- format_data(M2qc_disc_subset_top, type="exposure",
                                 header = TRUE,
                                 # phenotype_col = "",
                                 snp_col = "VariantID",
                                 beta_col = "BETA_FIXED",
                                 se_col = "SE_FIXED",
                                 eaf_col = "CAF",
                                 effect_allele_col = "CODEDALLELE",
                                 other_allele_col = "OTHERALLELE",
                                 pval_col = "P_FIXED",
                                 # gene_col = "NEAREST_GENE",
                                 chr_col = "Chr",
                                 pos_col = "BP",
                                 z_col = "Z_FIXED",
                                 samplesize_col = "N_EFF")

M2qc_disc_exp_dat_top$exposure <- "FABP4 (discovery, model 2)"
DT::datatable(M2qc_disc_exp_dat_top)

M3qc_disc_exp_dat_top <- format_data(M3qc_disc_subset_top, type="exposure",
                                 header = TRUE,
                                 # phenotype_col = "",
                                 snp_col = "VariantID",
                                 beta_col = "BETA_FIXED",
                                 se_col = "SE_FIXED",
                                 eaf_col = "CAF",
                                 effect_allele_col = "CODEDALLELE",
                                 other_allele_col = "OTHERALLELE",
                                 pval_col = "P_FIXED",
                                 # gene_col = "NEAREST_GENE",
                                 chr_col = "Chr",
                                 pos_col = "BP",
                                 z_col = "Z_FIXED",
                                 samplesize_col = "N_EFF")

M3qc_disc_exp_dat_top$exposure <- "FABP4 (discovery, model 3)"
DT::datatable(M3qc_disc_exp_dat_top)

M1qc_repl_exp_dat_top <- format_data(M1qc_repl_subset_top, type="exposure",
                                 header = TRUE,
                                 # phenotype_col = "",
                                 snp_col = "VariantID",
                                 beta_col = "BETA_FIXED",
                                 se_col = "SE_FIXED",
                                 eaf_col = "CAF",
                                 effect_allele_col = "CODEDALLELE",
                                 other_allele_col = "OTHERALLELE",
                                 pval_col = "P_FIXED",
                                 chr_col = "Chr",
                                 pos_col = "BP",
                                 z_col = "Z_FIXED",
                                 samplesize_col = "N")

M1qc_repl_exp_dat_top$exposure <- "FABP4 (replication)"
DT::datatable(M1qc_repl_exp_dat_top)

```

### Harmonization

We need to harmonize the data between exposure and outcomes
```{r Harmonize One2Many}
M1qc_disc_dat_top <- harmonise_data(exposure_dat = M1qc_disc_exp_dat_top, outcome_dat = outcome_dat)
DT::datatable(M1qc_disc_dat_top)

M2qc_disc_dat_top <- harmonise_data(exposure_dat = M2qc_disc_exp_dat_top, outcome_dat = outcome_dat)
DT::datatable(M2qc_disc_dat_top)

M3qc_disc_dat_top <- harmonise_data(exposure_dat = M3qc_disc_exp_dat_top, outcome_dat = outcome_dat)
DT::datatable(M3qc_disc_dat_top)


M1qc_repl_dat_top <- harmonise_data(exposure_dat = M1qc_repl_exp_dat_top, outcome_dat = outcome_dat)
DT::datatable(M1qc_repl_dat_top)

```

### Analysis: model 1

Executing the one-to-many analysis.

```{r MR One2Many}

M1qc_disc_res_top <- mr(M1qc_disc_dat_top, method_list = c("mr_ivw"))
M1qc_disc_res_top
M1qc_disc_res_top <- split_outcome(M1qc_disc_res_top) # to keep the Y axis label clean we exclude the exposure ID labels from the exposure column 

M1qc_disc_res_top <- sort_1_to_many(M1qc_disc_res_top, b = "b", sort_action = 4) #this sorts results by decreasing effect size (largest effect at top of the plot)

M1qc_repl_res_top <- mr(M1qc_repl_dat_top, method_list = c("mr_ivw"))
M1qc_repl_res_top
M1qc_repl_res_top <- split_outcome(M1qc_repl_res_top) # to keep the Y axis label clean we exclude the exposure ID labels from the exposure column 

M1qc_repl_res_top <- sort_1_to_many(M1qc_repl_res_top, b = "b", sort_action = 4) #this sorts results by decreasing effect size (largest effect at top of the plot)
```

### Analysis: model 2 and 3

Executing the one-to-many analysis.

```{r MR One2Many models}

M2qc_disc_res_top <- mr(M2qc_disc_dat_top, method_list = c("mr_ivw"))
M2qc_disc_res_top
M2qc_disc_res_top <- split_outcome(M2qc_disc_res_top) # to keep the Y axis label clean we exclude the exposure ID labels from the exposure column 

M2qc_disc_res_top <- sort_1_to_many(M2qc_disc_res_top, b = "b", sort_action = 4) #this sorts results by decreasing effect size (largest effect at top of the plot)

M3qc_disc_res_top <- mr(M3qc_disc_dat_top, method_list = c("mr_ivw"))
M3qc_disc_res_top
M3qc_disc_res_top <- split_outcome(M3qc_disc_res_top) # to keep the Y axis label clean we exclude the exposure ID labels from the exposure column 

M3qc_disc_res_top <- sort_1_to_many(M3qc_disc_res_top, b = "b", sort_action = 4) #this sorts results by decreasing effect size (largest effect at top of the plot)

```

### Visualization: model 1

```{r}
source(paste0(RESULTS,"/metagwas_fabp4/SCRIPTS/functions_mr.R"))
```

```{r warning=FALSE}
library(patchwork)
plot1 <- forest_plot_1_to_many(M1qc_disc_res_top, b = "b", se = "se",
                               exponentiate = TRUE, trans="log2", ao_slc = FALSE, 
                               # lo = min(M1qc_disc_res_top$b)-0.1,
                               # up = max(M1qc_disc_res_top$b)+0.2,
                               col1_width = 1, 
                               TraitM="outcome",
                               colour_scheme = uithof_color[3], 
                               col_text_size = 3, shape_points = 19)

plot2 <- forest_plot_1_to_many(M1qc_repl_res_top, b = "b", se = "se",
                               exponentiate = TRUE, trans="log2", ao_slc = FALSE, 
                               # lo = min(M1qc_disc_res_top$b)-0.1,
                               # up = max(M1qc_disc_res_top$b)+0.2,
                               col1_width = 1, 
                               TraitM="outcome",
                               colour_scheme = uithof_color[16], 
                               col_text_size = 3, shape_points = 19)

plot1  
plot2

pdf(paste0(PLOT_loc, "/",Today,"MR.results.disc.one2many.pdf"), height=10, width=8)
  plot1
dev.off() 

pdf(paste0(PLOT_loc, "/",Today,"MR.results.repl.one2many.pdf"), height=10, width=8)
  plot2
dev.off() 

```


### Visualization: model 2 and 3

```{r one2many models, warning=FALSE}
library(patchwork)
plot3 <- forest_plot_1_to_many(M2qc_disc_res_top, b = "b", se = "se",
                               exponentiate = TRUE, trans="log2", ao_slc = FALSE, 
                               # lo = min(M2qc_disc_res_top$b)-0.1,
                               # up = max(M2qc_disc_res_top$b)+0.2,
                               col1_width = 1, 
                               TraitM="outcome",
                               colour_scheme = uithof_color[9], 
                               col_text_size = 3, shape_points = 19)

plot4 <- forest_plot_1_to_many(M3qc_disc_res_top, b = "b", se = "se",
                               exponentiate = TRUE, trans="log2", ao_slc = FALSE, 
                               # lo = min(M3qc_disc_res_top$b)-0.1,
                               # up = max(M3qc_disc_res_top$b)+0.2,
                               col1_width = 1, 
                               TraitM="outcome",
                               colour_scheme = uithof_color[24], 
                               col_text_size = 3, shape_points = 19)

plot3  
plot4

pdf(paste0(ANALYSIS_loc, "/",Today,"MR.results.disc.m2.one2many.pdf"), height=10, width=8)
  plot3
dev.off() 

pdf(paste0(ANALYSIS_loc, "/",Today,"MR.results.disc.m3.one2many.pdf"), height=10, width=8)
  plot4
dev.off() 

```

# F-statistics

We will calculate the F-statistic which is consider a measure of the strength of the instruments; F > 10 is conderided a strong instrument. We will add these to the combined results

```{r Fstat}
#calculate F-statistic

cat("F-statistic for discovery instruments: \n")
Bdisc <- abs(M1qc_disc_dat_top$beta.exposure)
SEdisc <- M1qc_disc_dat_top$se.exposure

M1qc_disc_dat_top$Fdisc <- Bdisc^2/SEdisc^2
DT::datatable(M1qc_disc_dat_top)

cat("F-statistic for discovery model 2 instruments: \n")
Bdisc <- abs(M2qc_disc_dat_top$beta.exposure)
SEdisc <- M2qc_disc_dat_top$se.exposure

M2qc_disc_dat_top$Fdisc <- Bdisc^2/SEdisc^2
DT::datatable(M2qc_disc_dat_top)

cat("F-statistic for discovery model 3 instruments: \n")
Bdisc <- abs(M3qc_disc_dat_top$beta.exposure)
SEdisc <- M3qc_disc_dat_top$se.exposure

M3qc_disc_dat_top$Fdisc <- Bdisc^2/SEdisc^2
DT::datatable(M3qc_disc_dat_top)

cat("\nF-statistic for replication instruments: \n")
Bdisc <- abs(M1qc_repl_dat_top$beta.exposure)
SEdisc <- M1qc_repl_dat_top$se.exposure

M1qc_repl_dat_top$Fdisc <- Bdisc^2/SEdisc^2
DT::datatable(M1qc_repl_dat_top)

```

# Combine results

## Model 1: age + sex

```{r}

# discovery
instr_disc <- M1qc_disc_dat_top
res <- mr(M1qc_disc_dat_top, method_list = "mr_ivw")
het <- mr_heterogeneity(M1qc_disc_dat_top, method_list = c("mr_egger_regression", "mr_ivw"))
plt <- mr_pleiotropy_test(M1qc_disc_dat_top)
sin <- mr_singlesnp(M1qc_disc_dat_top, single_method = "mr_ivw")
loo <- mr_leaveoneout(M1qc_disc_dat_top, method = mr_ivw)

library(openxlsx)

## Create a new workbook
wb <- createWorkbook(title = "TwoSampleMR FABP4",
                     creator = "Sander W. van der Laan",
                     subject = "Causal inference")

## Add 3 worksheets
addWorksheet(wb, "MR_disc", tabColour = uithof_color[1])
addWorksheet(wb, "Het_disc", tabColour = uithof_color[2])
addWorksheet(wb, "Pleio_disc", tabColour = uithof_color[3])
addWorksheet(wb, "Single_disc", tabColour = uithof_color[4])
addWorksheet(wb, "Loo_disc", tabColour = uithof_color[5])
addWorksheet(wb, "Instr_disc", tabColour = uithof_color[6])

## Need data on worksheet to see all headers and footers
writeData(wb, sheet = 1, res)
writeData(wb, sheet = 2, het)
writeData(wb, sheet = 3, plt)
writeData(wb, sheet = 4, sin)
writeData(wb, sheet = 5, loo)
writeData(wb, sheet = 6, instr_disc)

# replication

instr_repl <- M1qc_repl_dat_top
res_repl <- mr(M1qc_repl_dat_top, method_list = "mr_ivw")
het_repl <- mr_heterogeneity(M1qc_repl_dat_top, method_list = c("mr_egger_regression", "mr_ivw"))
plt_repl <- mr_pleiotropy_test(M1qc_repl_dat_top)
sin_repl <- mr_singlesnp(M1qc_repl_dat_top, single_method = "mr_ivw")
loo_repl <- mr_leaveoneout(M1qc_repl_dat_top, method = mr_ivw)

## Add worksheets
addWorksheet(wb, "MR_repl", tabColour = uithof_color[15])
addWorksheet(wb, "Het_repl", tabColour = uithof_color[16])
addWorksheet(wb, "Pleio_repl", tabColour = uithof_color[17])
addWorksheet(wb, "Single_repl", tabColour = uithof_color[18])
addWorksheet(wb, "Loo_repl", tabColour = uithof_color[19])
addWorksheet(wb, "Instr_repl", tabColour = uithof_color[20])

## Need data on worksheet to see all headers and footers
writeData(wb, sheet = 7, res_repl)
writeData(wb, sheet = 8, het_repl)
writeData(wb, sheet = 9, plt_repl)
writeData(wb, sheet = 10, sin_repl)
writeData(wb, sheet = 11, loo_repl)
writeData(wb, sheet = 12, instr_repl)

# check contents

wb

## Save workbook
## Not run: 
saveWorkbook(wb, paste0(OUT_loc, "/",Today,"MR.results.disc_repl.xlsx"),
             overwrite = TRUE)

rm(res, het, plt, sin, loo, instr_repl, instr_disc,
   res_repl, het_repl, plt_repl, sin_repl, loo_repl)

```

## Model 2 & model 3

model 2: model 1 + BMI
model 3: model 1 + eGFR

```{r}

# discovery model 2
instr_disc <- M2qc_disc_dat_top
res <- mr(M2qc_disc_dat_top, method_list = "mr_ivw")
het <- mr_heterogeneity(M2qc_disc_dat_top, method_list = c("mr_egger_regression", "mr_ivw"))
plt <- mr_pleiotropy_test(M2qc_disc_dat_top)
sin <- mr_singlesnp(M2qc_disc_dat_top, single_method = "mr_ivw")
loo <- mr_leaveoneout(M2qc_disc_dat_top, method = mr_ivw)

library(openxlsx)

## Create a new workbook
wb <- createWorkbook(title = "TwoSampleMR FABP4",
                     creator = "Sander W. van der Laan",
                     subject = "Causal inference")

## Add 3 worksheets
addWorksheet(wb, "MR_disc_m2", tabColour = uithof_color[1])
addWorksheet(wb, "Het_disc_m2", tabColour = uithof_color[2])
addWorksheet(wb, "Pleio_disc_m2", tabColour = uithof_color[3])
addWorksheet(wb, "Single_disc_m2", tabColour = uithof_color[4])
addWorksheet(wb, "Loo_disc_m2", tabColour = uithof_color[5])
addWorksheet(wb, "Instr_disc_m2", tabColour = uithof_color[6])

## Need data on worksheet to see all headers and footers
writeData(wb, sheet = 1, res)
writeData(wb, sheet = 2, het)
writeData(wb, sheet = 3, plt)
writeData(wb, sheet = 4, sin)
writeData(wb, sheet = 5, loo)
writeData(wb, sheet = 6, instr_disc)

# discovery model 3
instr_disc_m3 <- M3qc_disc_dat_top
res_m3 <- mr(M3qc_disc_dat_top, method_list = "mr_ivw")
het_m3 <- mr_heterogeneity(M3qc_disc_dat_top, method_list = c("mr_egger_regression", "mr_ivw"))
plt_m3 <- mr_pleiotropy_test(M3qc_disc_dat_top)
sin_m3 <- mr_singlesnp(M3qc_disc_dat_top, single_method = "mr_ivw")
loo_m3 <- mr_leaveoneout(M3qc_disc_dat_top, method = mr_ivw)

## Add 3 worksheets
addWorksheet(wb, "MR_disc_m3", tabColour = uithof_color[1])
addWorksheet(wb, "Het_disc_m3", tabColour = uithof_color[2])
addWorksheet(wb, "Pleio_disc_m3", tabColour = uithof_color[3])
addWorksheet(wb, "Single_disc_m3", tabColour = uithof_color[4])
addWorksheet(wb, "Loo_disc_m3", tabColour = uithof_color[5])
addWorksheet(wb, "Instr_disc_m3", tabColour = uithof_color[6])

## Need data on worksheet to see all headers and footers
writeData(wb, sheet = 7, res_m3)
writeData(wb, sheet = 8, het_m3)
writeData(wb, sheet = 9, plt_m3)
writeData(wb, sheet = 10, sin_m3)
writeData(wb, sheet = 11, loo_m3)
writeData(wb, sheet = 12, instr_disc_m3)

# check contents

wb

## Save workbook
## Not run: 
saveWorkbook(wb, paste0(OUT_loc, "/",Today,"MR.results.disc_m2_m3.xlsx"),
             overwrite = TRUE)

rm(res, het, plt, sin, loo, instr_disc_m3, instr_disc_m2,
   res_m3, het_m3, plt_m3, sin_m3, loo_m3)

```




# Session information

------
  
    Version:      v1.0.3
    Last update:  2021-04-08
    Written by:   Sander W. van der Laan (s.w.vanderlaan-2[at]umcutrecht.nl).
    Description:  Script to get perform TwoSample MR.
    Minimum requirements: R version 3.5.2 (2018-12-20) -- 'Eggshell Igloo', macOS Mojave (10.14.2).
    
    Versions
    * v1.0.3 Updat to the color scheme for TwoSample MR on model 2 and 3.
    * v1.0.2 Add TwoSample MR for model 2 and model 3. 
    * v1.0.1 Add F-statistics to summary results.
    * v1.0.0 Initial version.

------
  
```{r eval = TRUE}
sessionInfo()
```

```{r clear intermediates}

rm(M1qc_disc, M1qc_repl)

```

```{r Saving}
save.image(paste0(PROJECT_loc, "/","TwoSampleMR.RData"))
```

------

<sup>&copy; 1979-2021 Sander W. van der Laan | s.w.vanderlaan-2[at]umcutrecht.nl | [swvanderlaan.github.io](https://swvanderlaan.github.io).</sup>
------
  
  