---
title: "R Notebook"
output: html_notebook
---

This is an [R Markdown](http://rmarkdown.rstudio.com) Notebook. When you execute code within the notebook, the results appear beneath the code. 

Try executing this chunk by clicking the *Run* button within the chunk or by placing your cursor inside it and pressing *Cmd+Shift+Enter*. 

```{r tidyCounts}
bulkRNA_counts %<>%
     drop_na(chr) %>%   # remove rows that have no information of start, end, chromosome and/or strand
     dplyr::select(1:6, one_of(sort(bulkRNA_meta$study_number))) # select gene expression of only patients in RNA-seq AE df, sort in same order as metadata study_number

# Cut up bulkRNA_counts into 'assay' and 'ranges' part
counts <- as.data.frame(bulkRNA_counts[,-(1:6)])              ## assay part
rownames(counts) <- bulkRNA_counts$ensembl_gene_id               ## assign rownames
rowRanges <- GRanges(bulkRNA_counts$chr,					## construct a GRanges object containing 4 columns (seqnames, ranges, strand, seqinfo) plus a metadata colum (feature_id): this will be the 'rowRanges' bit
                     IRanges(bulkRNA_counts$start, bulkRNA_counts$end),
                     strand = bulkRNA_counts$strand,
                     feature_id = bulkRNA_counts$ensembl_gene_id) #, df$pid)

# combine meta data from experiment with clinical data
bulkRNA_meta_clin <- merge(bulkRNA_meta, AEDB, by.x = "study_number", by.y = "study_number",
                           sort = FALSE, all.x = TRUE)

bulkRNA_meta_clin %<>%
  # mutate(macrophages = factor(macrophages, levels = c("no staining", "minor staining", "moderate staining", "heavy staining"))) %>% 
  # mutate(smc = factor(smc, levels = c("no staining", "minor staining", "moderate staining", "heavy staining"))) %>% 
  # mutate(calcification = factor(calcification, levels = c("no staining", "minor staining", "moderate staining", "heavy staining"))) %>% 
  # mutate(collagen = factor(collagen, levels = c("no staining", "minor staining", "moderate staining", "heavy staining"))) %>% 
  # mutate(fat = factor(fat, levels = c("no fat", "< 40% fat", "> 40% fat"))) %>% 
  mutate(study_number_row = study_number) %>%
  as.data.frame() %>%
  column_to_rownames("study_number_row")

head(bulkRNA_meta_clin)
dim(bulkRNA_meta_clin)

```

### SummarizedExperiment

Here we create a container that brings all data together.
```{r se}
(AERNASE <- SummarizedExperiment(assays = list(counts = as.matrix(counts)),  ## as.matrix() is important here, to get dimnames the same (?) - see error below if you don't use this
                            rowRanges = rowRanges, 
                            colData = bulkRNA_meta_clin,
                            metadata = "Athero-Express Biobank Study bulk RNA sequencing. Sample type: carotid plaques. Technology: CEL2-seq adapted for bulk RNA sequencing, thus 3'-focused.")   )
```

Add a new chunk by clicking the *Insert Chunk* button on the toolbar or by pressing *Cmd+Option+I*.

When you save the notebook, an HTML file containing the code and output will be saved alongside it (click the *Preview* button or press *Cmd+Shift+K* to preview the HTML file). 

The preview shows you a rendered HTML copy of the contents of the editor. Consequently, unlike *Knit*, *Preview* does not run any R code chunks. Instead, the output of the chunk when it was last run in the editor is displayed.

